pkg/ipcache/cidr.go

```diff
@@ -29,12 +29,40 @@ import (
 // allocation fails, all allocations are rolled back and the error is returned.
 // When an identity is freshly allocated for a CIDR, it is added to the
 // ipcache.
-func AllocateCIDRs(impl Implementation, prefixes []*net.IPNet) error {
+func AllocateCIDRs(impl Implementation, prefixes []*net.IPNet) ([]*identity.Identity, error) {
 	// First, if the implementation will complain, exit early.
 	if err := checkPrefixes(impl, prefixes); err != nil {
-		return err
+		return nil, err
 	}
 
+	return allocateCIDRs(prefixes)
+}
+
+// AllocateCIDRsForIPs attempts to allocate identities for a list of CIDRs. It
+// fails if any of the CIDRs do not correspond to /32 or /128 IP addresses. If
+// any allocation fails, all allocations are rolled back and the error is
+// returned. When an identity is freshly allocated for a CIDR, it is added to
+// the ipcache.
+func AllocateCIDRsForIPs(prefixes []*net.IPNet) ([]*identity.Identity, error) {
```

Comment:
Why does this take `[]*net.IPNet` and not `[]*net.IP`?

Just wondering if there's a way to avoid the prefix validation below since it seems like the caller is guaranteed to have real IPs and we should be able to programmatically generate full masks (`/32` for IPv4, `/128` for IPv6).

Reply:
This would indeed be simpler :) fixed!